include/agent.h

```diff
@@ -0,0 +1,96 @@
+#ifndef _ABM_AGENT_H_
+#define _ABM_AGENT_H_
+
+#include <memory>
+#include <vector>
+
+#include "config.h"
+#include "graph.h"
+#include "agent.h"
+
+namespace abm {
+
+//! Agent class that has id, origin, destination, departure time, current node and compute path from current node to destination.
+class Agent {
+ public:
+  //! Construct an agent class
+  //! \param[in] graph Pass a reference of the graph that the agent does routing on
+  explicit Agent(const std::shared_ptr<abm::Graph>& graph) : graph_{graph} {};
```

Comment:
A graph should not be passed as an input parameter to Agent, create an id instead or id, source and destination.

Reply:
1. Create an id as each agent is initialized?
2. Should all the graph part be moved to another class (comment out in the agent class now)?

Reply:
Graph should not be part of agent, please remove all graph parts

Reply:
Constructor should have an id, source and destination and add another constructor with just agent id